src/ipahealthcheck/ipa/kdc.py

```diff
@@ -0,0 +1,71 @@
+
+# Copyright (C) 2022 FreeIPA Contributors see COPYING for license
+#
+
+import logging
+import os
+
+from ipahealthcheck.ipa.plugin import IPAPlugin, registry
+from ipahealthcheck.core.plugin import Result, duration
+from ipahealthcheck.core import constants
+
+
+logger = logging.getLogger()
+SYSCONFIG = '/etc/sysconfig/krb5kdc'
+
+
+def get_contents(file):
+    with open(SYSCONFIG, "r") as fd:
+        lines = fd.readlines()
+    return lines
+
+
+@registry
+class KDCWorkersCheck(IPAPlugin):
+    """Verify that the number of workers matches the number of cores"""
+
+    @duration
+    def check(self):
+        key = 'workers'
+        cpus = os.sysconf('SC_NPROCESSORS_ONLN')
+        logging.debug('Detected %s CPUs', cpus)
+
+        lines = get_contents(SYSCONFIG)
+
+        args_read = False
+        for line in lines:
+            sline = line.strip()
+            workers = 0
+            if sline.startswith('KRB5KDC_ARGS'):
+                args_read = True
+                sline = sline.split('=', maxsplit=1)[1]
+                if sline.find("-w") == -1:
+                    yield Result(self, constants.WARNING, key=key,
+                                 sysconfig=SYSCONFIG,
+                                 msg='No KDC workers defined in '
+                                 '{sysconfig}')
+                    return
+
+                # Making an assumption that this line is not misconfigured
+                # otherwise the KDC wouldn't start at all.
+                sline = sline.replace("'", "")
```

Comment:
Nitpicking but if the file was manually edited, the line may contain double quotes instead of simple quotes and in this case the check would see 0 workers:
```
# ipa-healthcheck --source ipahealthcheck.ipa.kdc
[
  {
    "source": "ipahealthcheck.ipa.kdc",
    "check": "KDCWorkersCheck",
    "result": "WARNING",
    "uuid": "cf02d93d-235f-404c-88fd-dc842cea7b17",
    "when": "20220503143144Z",
    "duration": "0.000186",
    "kw": {
      "key": "workers",
      "cpus": 2,
      "workers": 0,
      "sysconfig": "/etc/sysconfig/krb5kdc",
      "msg": "The number of CPUs {cpus} does not match the number of workers {workers} in {sysconfig}"
    }
  }
]
# cat /etc/sysconfig/krb5kdc
KRB5REALM=IPA.TEST
KRB5KDC_ARGS="-w 4"
```